subsys/net/lib/nrf_cloud/src/nrf_cloud_transport.c

```diff
@@ -570,7 +570,7 @@ static void nct_mqtt_evt_handler(struct mqtt_client *const mqtt_client,
 		} else {
 			/* Try to match it with one of the data topics. */
 			dc.id = p->message_id;
-			dc.data.ptr = p->message.payload.data;
+			dc.data.ptr = nct.payload_buf;
```

Comment:
Nice catch

Reply:
If it's a bugfix, it should be in a separate commit.